SetFlags.cmake

```diff
@@ -233,87 +81,76 @@ macro(enable_profile)
-			endif()
-			if ("${CLANG_VERSION}" VERSION_GREATER 3.4)
-				add_flags_cxx("-Wno-error=disabled-macro-expansion")
-			endif()
-			if ("${CLANG_VERSION}" VERSION_GREATER 3.5)
-				include(CheckCXXCompilerFlag)
-				check_cxx_compiler_flag(-Wno-reserved-id-macro HAS_NO_RESERVED_ID_MACRO)
-				check_cxx_compiler_flag(-Wno-documentation-unknown-command HAS_NO_DOCUMENTATION_UNKNOWN)
-				if (HAS_NO_RESERVED_ID_MACRO)
-					# Use this flag to ignore error for a reserved macro problem in sqlite 3
-					add_flags_cxx("-Wno-reserved-id-macro")
-				endif()
-				if (HAS_NO_DOCUMENTATION_UNKNOWN)
-					# Ignore another problem in sqlite
-					add_flags_cxx("-Wno-documentation-unknown-command")
-				endif()
-			endif()
-			if ("${CLANG_VERSION}" VERSION_GREATER 3.7)
-				check_cxx_compiler_flag(-Wno-double-promotion HAS_NO_DOUBLE_PROMOTION)
-				if (HAS_NO_DOUBLE_PROMOTION)
-					add_flags_cxx("-Wno-double-promotion")
-				endif()
-			endif()
-			add_flags_cxx("-Wno-error=unused-command-line-argument")
-			add_flags_cxx("-Wno-documentation-unknown-command")
-		endif()
+function(enable_warnings TARGET)
+	if (MSVC)
+		# TODO: MSVC level 4, warnings as errors
+		return ()
 	endif()
 
-endmacro()
+	target_compile_options(
+		${TARGET} PRIVATE
+
+		# We use a signed char (fixes #640 on RasPi)
+		# TODO: specify this in code, not a compile flag:
+		-fsigned-char
+
+		# We support non-IEEE 754 FPUs so can make no guarantees about error:
+		-ffast-math
 
-#	if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-#		foreach(FILENAME ${ARGN})
-#			message("downgrade_warnings for ${FILENAME}")
-#			set_source_files_properties(${FILENAME} PROPERTIES COMPILE_FLAGS "-Wno-error=sign-conversion -Wno-error=conversion")
-#			set_source_files_properties(${FILENAME} PROPERTIES COMPILE_FLAGS "-Wno-error=missing-prototypes -Wno-error=deprecated")
-#			set_source_files_properties(${FILENAME} PROPERTIES COMPILE_FLAGS "-Wno-error=shadow -Wno-error=old-style-cast  -Wno-error=switch-enum -Wno-error=switch")
-#			set_source_files_properties(${FILENAME} PROPERTIES COMPILE_FLAGS "-Wno-error=float-equal -Wno-error=global-constructors")
+		# All warnings:
+		-Wall -Wextra
 
-#			if ("${CLANG_VERSION}" VERSION_GREATER 3.0)
-#				# flags that are not present in 3.0
-#				set_source_files_properties(${FILENAME} PROPERTIES COMPILE_FLAGS "-Wno-error=covered-switch-default ")
-#				set_source_files_properties(${FILENAME} PROPERTIES COMPILE_FLAGS "-Wno-implicit-fallthrough -Wno-error=extra-semi")
-#				set_source_files_properties(${FILENAME} PROPERTIES COMPILE_FLAGS "-Wno-error=missing-variable-declarations")
-#			endif()
-#		endforeach()
-#	endif()
+		# TODO: actually fix the warnings instead of disabling them
+		# or at least disable on a file-level basis:
+		-Wno-unused-parameter -Wno-missing-noreturn -Wno-padded -Wno-implicit-fallthrough
+		-Wno-double-promotion
+
+		# This is a pretty useless warning, we've already got -Wswitch which is what we need:
+		-Wno-switch-enum
```

Comment:
These being `-Wno` was somewhat justified for clang, since we want to be `-Werror` clean. Skipping it for gcc as well makes no sense though.